### PR TITLE
Add more specs, improve some existing ones

### PR DESCRIPTION
**Pull request by jeffcarbs**
Moved specs from ZP and improved some of the existing specs.

### CODE REVIEW

File: lib/fixy/formatter/rate.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ module Rate
       #
 
       def format_rate(input, length)
-        raise ArgumentError, "Invalid rate (rate: #{input}. Rate must be >= 0." if input < 0
+        raise ArgumentError, "Invalid rate (rate: #{input}). Rate must be >= 0." if input < 0
```
Comment (CodyWatters) [inline lib/fixy/formatter/rate.rb:17]: nice

File: spec/fixy/formatter/alphanumeric_spec.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+require 'spec_helper'
+
+describe Fixy::Formatter::Alphanumeric do
+  let(:proxy) do
+    Class.new do
+      include Fixy::Formatter::Alphanumeric
+
+      self::LINE_ENDING_CRLF = "\r\n"
```
Comment (CodyWatters) [inline spec/fixy/formatter/alphanumeric_spec.rb:8]: why windows return? doesn't really matter but curious

File: spec/fixy/formatter/amount_spec.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  end
+
+  describe '#format_amount' do
+    subject { proxy.format_amount(input, length) }
+
+    let(:length) { 10 }
+
+    [
+      # [nil,     3,  '000'],
+      # ['',      3,  '000'],
```
Comment (CodyWatters) [inline spec/fixy/formatter/amount_spec.rb:18]: You could re-add these if https://github.com/Gusto/fixy/blob/master/lib/fixy/formatter/amount.rb#L17 was updated from
```ruby
value = (('%0' << length.to_s << 'd') % integerize(input.abs * 100))
```
to
```ruby
value = (('%0' << length.to_s << 'd') % integerize(input.to_f.abs * 100))
```

File: spec/fixy/formatter/rate_spec.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  end
+
+  describe '#format_rate' do
+    subject { proxy.format_rate(input, length) }
+
+    let(:length) { 10 }
+
+    [
+      # [nil,       4, '0000'],
+      # ['',        4, '0000'],
```
Comment (CodyWatters) [inline spec/fixy/formatter/rate_spec.rb:17]: You could uncomment these if you updated
https://github.com/Gusto/fixy/blob/master/lib/fixy/formatter/rate.rb#L18
from
```ruby
(('%0' << length.to_s << 'd') % integerize(input * (10 ** length)))[0..(length - 1)]
```
to
```ruby
(('%0' << length.to_s << 'd') % integerize(input.to_f * (10 ** length)))[0..(length - 1)]
```

File: spec/fixy/formatter/signed_amount_spec.rb [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  end
+
+  describe '#format_signed_amount' do
+    subject { proxy.format_signed_amount(input, length) }
+
+    let(:length) { 10 }
+
+    [
+      # [nil,     6,  '00000+'],
+      # ['',      6,  '00000+'],
```
Comment (CodyWatters) [inline spec/fixy/formatter/signed_amount_spec.rb:17]: You could uncomment these if you update
https://github.com/Gusto/fixy/blob/master/lib/fixy/formatter/signed_amount.rb#L14
from
```ruby
sign = input >= 0 ? '+' : '-'
```
to
```ruby
sign = input.to_f >= 0 ? '+' : '-'
```

**Review by CodyWatters — APPROVED** [review]
looks good - I think it might be worth updating those empty string / nil specs to pass since you're adding all these specs anyway. There may be something worthwhile in having those specs fail for empty strings and nils - but idk what that would be. @zp-alfred accept

**Comment by zp-alfred** [comment]
Accepted by CodyWatters!

**Comment by zp-alfred** [comment]
Accepted by CodyWatters!

**Comment by zp-alfred** [comment]
Accepted by CodyWatters!

**Comment by zp-alfred** [comment]
Accepted by CodyWatters!

File: spec/fixy/formatter/alphanumeric_spec.rb (hunk 2 above)
Comment (jeffcarbs) [reply spec/fixy/formatter/alphanumeric_spec.rb:8]: Not exactly sure, this was following another spec's example. You needed to specify the LINE_ENDING_CRLF to use the alphanumeric formatter.

**Comment by jeffcarbs** [comment]
@CodyWatters - good points on using `to_f` to be able to pass nil or empty string. The reason those tests are there and commented out is because the `Numeric` formatter actually handles `nil` and `''` and I thought these would too. In turns out they didn't but I'm not sure if that's a feature or a bug 🤔  so I think for now we should leave them to not change any behavior.

**Pull request merged**